src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -146,8 +143,7 @@ internal static OtlpTrace.Span ToOtlpSpan(this Activity activity)
                     continue;
                 }
 
-                // TODO: Retrieve resources for trace (not library/ActivitySource)
-                var resource = fakeResource;
+                var resource = activity.GetResource() ?? Resource.Empty;
```

Comment:
Empty is returned by GetResource is null, so do you need the null check here?

Reply:
no, we strictly don't. I can remove that bit